spring-cloud-gcp-samples/spring-cloud-gcp-secretmanager-sample/src/main/java/com/example/SecretManagerWebController.java

```diff
@@ -46,25 +48,58 @@
 	@Value("${secrets.application-secret}")
 	private String applicationSecretValue;
 
+	// Application secret is set into the properties file and get here using @Value
+	@Value("${my-application-secret}")
+	private String myApplicationSecretValue;
+
 	@GetMapping("/")
 	public ModelAndView renderIndex(ModelMap map) {
 		map.put("applicationSecret", this.applicationSecretValue);
+		map.put("myApplicationSecret", this.myApplicationSecretValue);
 		return new ModelAndView("index.html", map);
 	}
 
 	@GetMapping("/getSecret")
 	@ResponseBody
-	public String getSecret(@RequestParam String secretId, ModelMap map) {
-		String secretPayload = this.secretManagerTemplate.getSecretString(secretId);
+	public String getSecret(
+			@RequestParam String secretId,
+			@RequestParam(required = false) String version,
+			@RequestParam(required = false) String projectId,
+			ModelMap map) {
+
+		if (StringUtils.isEmpty(version)) {
+			version = SecretManagerTemplate.LATEST_VERSION;
+		}
+
+		String secretPayload;
+		if (StringUtils.isEmpty(projectId)) {
+			secretPayload = this.secretManagerTemplate.getSecretString(secretId, version);
```

Comment:
Should we remove this variant of `getSecretString` as well? Same goes for `getSecretBytes`.

Reply:
These variants exists in the version 1.2.2. We shouldn't remove them to ensure the downward compatibility.

Reply:
Ah, didn't realize that. Thanks!